File: src/domain/tasks/task_entity.go
package tasks

type (
	Task struct {
		ID          int64  `sql:"primary_key" json:"id"`
		UserID      int64  `sql:"user_id" json:"userId"`
		Content     string `sql:"content" json:"content"`
		CreatedDate string `sql:"created_date,default:CURRENT_TIMESTAMP" json:"createdDate"`
	}

	Tasks []Task
)

File: src/features/task/entities/task.entity.ts
import { BaseEntity, Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from "typeorm";

@Entity({ name: 'tasks' })
export class TaskEntity extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    content: string;

    @Column()
    userId: number;

    @CreateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP(6)" })
    createdAt: Date;

    @UpdateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP(6)", onUpdate: "CURRENT_TIMESTAMP(6)" })
    updatedAt: Date;
}
